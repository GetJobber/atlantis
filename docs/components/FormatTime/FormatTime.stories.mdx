import { Canvas, Meta } from "@storybook/addon-docs";
import { CivilTime } from "@std-proposal/temporal";
import { FormatTime } from "@jobber/components/FormatTime";

<Meta title="Components/Utilities/FormatTime" />

# FormatTime

In Jobber a FormatTime is used to ensure that time is displayed in the expected
format. No text styling is applied, this simply formats the text.

## Specify format

You can select a specific format to use in FormatTime. See the examples below:

<Canvas>
  <div>
    <FormatTime time={new CivilTime(2, 35)} use24HourClock={true} />
    -
    <FormatTime time={new CivilTime(18, 35)} use24HourClock={true} />
    <br />
    <FormatTime time={new CivilTime(2, 35)} use24HourClock={false} />
    -
    <FormatTime time={new CivilTime(18, 35)} use24HourClock={false} />
  </div>
</Canvas>
