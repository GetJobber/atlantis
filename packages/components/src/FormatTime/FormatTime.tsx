import React from "react";
import { CivilTime } from "@std-proposal/temporal";

interface FormatTimeProps {
  /**
   * Civil Time of time to be displayed.
   */
  readonly time: CivilTime;

  /**
   * Optionally specify clock format. If `undefined` system format will be respected.
   */
  readonly use24HourClock?: boolean;
}

export function FormatTime({ time, use24HourClock }: FormatTimeProps) {
  return <>{formatCivilTime(time, use24HourClock)}</>;
}

function formatCivilTime(time: CivilTime, use24HourClock?: boolean) {
  const currentDate = new Date();
  const currentYear = currentDate.getFullYear();
  const currentMonth = currentDate.getMonth();
  const currentDay = currentDate.getDay();
  const date = new Date(
    currentYear,
    currentMonth,
    currentDay,
    time.hour,
    time.minute,
    time.second,
    time.millisecond,
  );

  return date.toLocaleTimeString(navigator.language, {
    hour12: typeof use24HourClock === "boolean" ? !use24HourClock : undefined,
    minute: "2-digit",
    hour: "numeric",
  });
}
