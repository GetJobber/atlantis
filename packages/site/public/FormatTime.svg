<svg width="224" height="117" viewBox="0 0 224 117" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_27_6364)">
<path d="M40 47C40 42.5817 43.5817 39 48 39H177C181.418 39 185 42.5817 185 47V69C185 73.4183 181.418 77 177 77H48C43.5817 77 40 73.4183 40 69V47Z" fill="#F1F0E9"/>
<path d="M57.7006 53.496L58.3486 54.528L52.7006 57.992L58.3486 61.488L57.6286 62.472L51.5486 58.672V57.296L57.7006 53.496ZM64.5344 51.808C65.7237 51.808 66.6224 52.2987 67.2304 53.28C67.8384 54.2613 68.1424 55.664 68.1424 57.488C68.1424 59.3173 67.8384 60.7253 67.2304 61.712C66.6224 62.6933 65.7237 63.184 64.5344 63.184C63.3504 63.184 62.4517 62.6933 61.8384 61.712C61.2304 60.7253 60.9264 59.3173 60.9264 57.488C60.9264 55.664 61.2304 54.2613 61.8384 53.28C62.4517 52.2987 63.3504 51.808 64.5344 51.808ZM64.5344 52.888C63.857 52.888 63.3237 53.248 62.9344 53.968C62.545 54.6827 62.3504 55.856 62.3504 57.488C62.3504 58.3147 62.393 59.0213 62.4784 59.608C62.5637 60.1947 62.6944 60.672 62.8704 61.04C63.0517 61.4027 63.2784 61.6693 63.5504 61.84C63.8277 62.0107 64.1557 62.096 64.5344 62.096C64.993 62.096 65.385 61.9387 65.7104 61.624C66.041 61.3093 66.2917 60.8133 66.4624 60.136C66.6384 59.4533 66.7264 58.5707 66.7264 57.488C66.7264 56.3893 66.649 55.5013 66.4944 54.824C66.345 54.1467 66.1077 53.656 65.7824 53.352C65.4624 53.0427 65.0464 52.888 64.5344 52.888ZM66.1584 53.28L66.5104 54.456L63.0064 61.552L62.5104 60.616L66.1584 53.28ZM74.1281 51.808C75.3175 51.808 76.2161 52.2987 76.8241 53.28C77.4321 54.2613 77.7361 55.664 77.7361 57.488C77.7361 59.3173 77.4321 60.7253 76.8241 61.712C76.2161 62.6933 75.3175 63.184 74.1281 63.184C72.9441 63.184 72.0455 62.6933 71.4321 61.712C70.8241 60.7253 70.5201 59.3173 70.5201 57.488C70.5201 55.664 70.8241 54.2613 71.4321 53.28C72.0455 52.2987 72.9441 51.808 74.1281 51.808ZM74.1281 52.888C73.4508 52.888 72.9175 53.248 72.5281 53.968C72.1388 54.6827 71.9441 55.856 71.9441 57.488C71.9441 58.3147 71.9868 59.0213 72.0721 59.608C72.1575 60.1947 72.2881 60.672 72.4641 61.04C72.6455 61.4027 72.8721 61.6693 73.1441 61.84C73.4215 62.0107 73.7495 62.096 74.1281 62.096C74.5868 62.096 74.9788 61.9387 75.3041 61.624C75.6348 61.3093 75.8855 60.8133 76.0561 60.136C76.2321 59.4533 76.3201 58.5707 76.3201 57.488C76.3201 56.3893 76.2428 55.5013 76.0881 54.824C75.9388 54.1467 75.7015 53.656 75.3761 53.352C75.0561 53.0427 74.6401 52.888 74.1281 52.888ZM75.7521 53.28L76.1041 54.456L72.6001 61.552L72.1041 60.616L75.7521 53.28ZM82.4179 60.6C82.4179 60.2533 82.5405 59.952 82.7859 59.696C83.0365 59.44 83.3432 59.312 83.7059 59.312C84.0792 59.312 84.3885 59.44 84.6339 59.696C84.8845 59.952 85.0099 60.2533 85.0099 60.6C85.0099 60.9627 84.8845 61.272 84.6339 61.528C84.3885 61.7787 84.0792 61.904 83.7059 61.904C83.3432 61.904 83.0365 61.7787 82.7859 61.528C82.5405 61.272 82.4179 60.9627 82.4179 60.6ZM82.4179 54.832C82.4179 54.5867 82.4739 54.368 82.5859 54.176C82.6979 53.984 82.8525 53.832 83.0499 53.72C83.2472 53.6027 83.4659 53.544 83.7059 53.544C84.0792 53.544 84.3885 53.6693 84.6339 53.92C84.8845 54.1707 85.0099 54.4747 85.0099 54.832C85.0099 55.184 84.8845 55.4907 84.6339 55.752C84.3885 56.008 84.0792 56.136 83.7059 56.136C83.3432 56.136 83.0365 56.008 82.7859 55.752C82.5405 55.4907 82.4179 55.184 82.4179 54.832ZM93.3156 51.808C94.505 51.808 95.4036 52.2987 96.0116 53.28C96.6196 54.2613 96.9236 55.664 96.9236 57.488C96.9236 59.3173 96.6196 60.7253 96.0116 61.712C95.4036 62.6933 94.505 63.184 93.3156 63.184C92.1316 63.184 91.233 62.6933 90.6196 61.712C90.0116 60.7253 89.7076 59.3173 89.7076 57.488C89.7076 55.664 90.0116 54.2613 90.6196 53.28C91.233 52.2987 92.1316 51.808 93.3156 51.808ZM93.3156 52.888C92.6383 52.888 92.105 53.248 91.7156 53.968C91.3263 54.6827 91.1316 55.856 91.1316 57.488C91.1316 58.3147 91.1743 59.0213 91.2596 59.608C91.345 60.1947 91.4756 60.672 91.6516 61.04C91.833 61.4027 92.0596 61.6693 92.3316 61.84C92.609 62.0107 92.937 62.096 93.3156 62.096C93.7743 62.096 94.1663 61.9387 94.4916 61.624C94.8223 61.3093 95.073 60.8133 95.2436 60.136C95.4196 59.4533 95.5076 58.5707 95.5076 57.488C95.5076 56.3893 95.4303 55.5013 95.2756 54.824C95.1263 54.1467 94.889 53.656 94.5636 53.352C94.2436 53.0427 93.8276 52.888 93.3156 52.888ZM94.9396 53.28L95.2916 54.456L91.7876 61.552L91.2916 60.616L94.9396 53.28ZM102.909 51.808C104.099 51.808 104.997 52.2987 105.605 53.28C106.213 54.2613 106.517 55.664 106.517 57.488C106.517 59.3173 106.213 60.7253 105.605 61.712C104.997 62.6933 104.099 63.184 102.909 63.184C101.725 63.184 100.827 62.6933 100.213 61.712C99.6054 60.7253 99.3014 59.3173 99.3014 57.488C99.3014 55.664 99.6054 54.2613 100.213 53.28C100.827 52.2987 101.725 51.808 102.909 51.808ZM102.909 52.888C102.232 52.888 101.699 53.248 101.309 53.968C100.92 54.6827 100.725 55.856 100.725 57.488C100.725 58.3147 100.768 59.0213 100.853 59.608C100.939 60.1947 101.069 60.672 101.245 61.04C101.427 61.4027 101.653 61.6693 101.925 61.84C102.203 62.0107 102.531 62.096 102.909 62.096C103.368 62.096 103.76 61.9387 104.085 61.624C104.416 61.3093 104.667 60.8133 104.837 60.136C105.013 59.4533 105.101 58.5707 105.101 57.488C105.101 56.3893 105.024 55.5013 104.869 54.824C104.72 54.1467 104.483 53.656 104.157 53.352C103.837 53.0427 103.421 52.888 102.909 52.888ZM104.533 53.28L104.885 54.456L101.381 61.552L100.885 60.616L104.533 53.28ZM111.199 60.6C111.199 60.2533 111.322 59.952 111.567 59.696C111.818 59.44 112.124 59.312 112.487 59.312C112.86 59.312 113.17 59.44 113.415 59.696C113.666 59.952 113.791 60.2533 113.791 60.6C113.791 60.9627 113.666 61.272 113.415 61.528C113.17 61.7787 112.86 61.904 112.487 61.904C112.124 61.904 111.818 61.7787 111.567 61.528C111.322 61.272 111.199 60.9627 111.199 60.6ZM111.199 54.832C111.199 54.5867 111.255 54.368 111.367 54.176C111.479 53.984 111.634 53.832 111.831 53.72C112.028 53.6027 112.247 53.544 112.487 53.544C112.86 53.544 113.17 53.6693 113.415 53.92C113.666 54.1707 113.791 54.4747 113.791 54.832C113.791 55.184 113.666 55.4907 113.415 55.752C113.17 56.008 112.86 56.136 112.487 56.136C112.124 56.136 111.818 56.008 111.567 55.752C111.322 55.4907 111.199 55.184 111.199 54.832ZM122.097 51.808C123.286 51.808 124.185 52.2987 124.793 53.28C125.401 54.2613 125.705 55.664 125.705 57.488C125.705 59.3173 125.401 60.7253 124.793 61.712C124.185 62.6933 123.286 63.184 122.097 63.184C120.913 63.184 120.014 62.6933 119.401 61.712C118.793 60.7253 118.489 59.3173 118.489 57.488C118.489 55.664 118.793 54.2613 119.401 53.28C120.014 52.2987 120.913 51.808 122.097 51.808ZM122.097 52.888C121.42 52.888 120.886 53.248 120.497 53.968C120.108 54.6827 119.913 55.856 119.913 57.488C119.913 58.3147 119.956 59.0213 120.041 59.608C120.126 60.1947 120.257 60.672 120.433 61.04C120.614 61.4027 120.841 61.6693 121.113 61.84C121.39 62.0107 121.718 62.096 122.097 62.096C122.556 62.096 122.948 61.9387 123.273 61.624C123.604 61.3093 123.854 60.8133 124.025 60.136C124.201 59.4533 124.289 58.5707 124.289 57.488C124.289 56.3893 124.212 55.5013 124.057 54.824C123.908 54.1467 123.67 53.656 123.345 53.352C123.025 53.0427 122.609 52.888 122.097 52.888ZM123.721 53.28L124.073 54.456L120.569 61.552L120.073 60.616L123.721 53.28ZM131.691 51.808C132.88 51.808 133.779 52.2987 134.387 53.28C134.995 54.2613 135.299 55.664 135.299 57.488C135.299 59.3173 134.995 60.7253 134.387 61.712C133.779 62.6933 132.88 63.184 131.691 63.184C130.507 63.184 129.608 62.6933 128.995 61.712C128.387 60.7253 128.083 59.3173 128.083 57.488C128.083 55.664 128.387 54.2613 128.995 53.28C129.608 52.2987 130.507 51.808 131.691 51.808ZM131.691 52.888C131.013 52.888 130.48 53.248 130.091 53.968C129.701 54.6827 129.507 55.856 129.507 57.488C129.507 58.3147 129.549 59.0213 129.635 59.608C129.72 60.1947 129.851 60.672 130.027 61.04C130.208 61.4027 130.435 61.6693 130.707 61.84C130.984 62.0107 131.312 62.096 131.691 62.096C132.149 62.096 132.541 61.9387 132.867 61.624C133.197 61.3093 133.448 60.8133 133.619 60.136C133.795 59.4533 133.883 58.5707 133.883 57.488C133.883 56.3893 133.805 55.5013 133.651 54.824C133.501 54.1467 133.264 53.656 132.939 53.352C132.619 53.0427 132.203 52.888 131.691 52.888ZM133.315 53.28L133.667 54.456L130.163 61.552L129.667 60.616L133.315 53.28ZM153.014 60.192H148.654L147.774 63H146.406L149.99 51.984H151.766L155.35 63H153.894L153.014 60.192ZM149.014 59.04H152.654L150.846 53.152L149.014 59.04ZM164.16 51.984L164.784 63H163.448L163.104 57.248C163.088 56.8587 163.072 56.464 163.056 56.064C163.04 55.664 163.029 55.28 163.024 54.912C163.019 54.544 163.016 54.216 163.016 53.928C163.016 53.64 163.016 53.416 163.016 53.256L161.16 61.024H159.768L157.776 53.256C157.781 53.4107 157.787 53.64 157.792 53.944C157.797 54.2427 157.8 54.5813 157.8 54.96C157.805 55.3387 157.803 55.728 157.792 56.128C157.787 56.5227 157.776 56.8907 157.76 57.232L157.464 63H156.16L156.784 51.984H158.72L160.504 59.528L162.208 51.984H164.16ZM167.322 53.496L173.474 57.296V58.672L167.402 62.472L166.674 61.48L172.322 57.992L166.674 54.512L167.322 53.496Z" fill="#233D48"/>
</g>
<defs>
<clipPath id="clip0_27_6364">
<path d="M0 8C0 3.58172 3.58172 0 8 0H216C220.418 0 224 3.58172 224 8V109C224 113.418 220.418 117 216 117H8C3.58172 117 0 113.418 0 109V8Z" fill="white"/>
</clipPath>
</defs>
</svg>
