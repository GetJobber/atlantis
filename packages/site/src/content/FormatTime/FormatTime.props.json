[
  {
    "tags": {},
    "filePath": "packages/components/src/FormatTime/FormatTime.tsx",
    "description": "",
    "displayName": "FormatTime",
    "methods": [],
    "props": {
      "time": {
        "defaultValue": null,
        "description": "Time (as JS Date) to be displayed.\n\nA `string` should be an ISO 8601 format date string.",
        "name": "time",
        "parent": {
          "fileName": "packages/components/src/FormatTime/FormatTime.tsx",
          "name": "FormatTimeProps"
        },
        "required": true,
        "type": {
          "name": "string | Date"
        }
      },
      "use24HourClock": {
        "defaultValue": null,
        "description": "Optionally specify clock format. If `undefined` system format will be respected.",
        "name": "use24HourClock",
        "parent": {
          "fileName": "packages/components/src/FormatTime/FormatTime.tsx",
          "name": "FormatTimeProps"
        },
        "required": false,
        "type": {
          "name": "boolean"
        }
      }
    }
  }
]