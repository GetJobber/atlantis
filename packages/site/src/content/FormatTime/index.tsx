import FormatTimeContent from "@atlantis/docs/components/FormatTime/FormatTime.stories.mdx";
import Props from "./FormatTime.props.json";
import { ContentExport } from "../../types/content";

export default {
  content: () => <FormatTimeContent />,
  props: Props,
  component: {
    element: `const earlyTime = new Date();
  earlyTime.setHours(2, 35, 0, 0);

  const lateTime = new Date();
  lateTime.setHours(22, 35, 0, 0);

  return (
    <>
      <FormatTime time={earlyTime} /> - <FormatTime time={lateTime} />
    </>
  );`,
    defaultProps: { time: "2020-07-10 15:22:00.000" },
  },
  title: "FormatTime",
  links: [
    {
      label: "Storybook",
      url: `${
        (window as AtlantisWindow)?.env?.VITE_STORYBOOK_URL
      }/?path=/docs/components-utilities-FormatTime-web--docs`,
    },
  ],
} as const satisfies ContentExport;
