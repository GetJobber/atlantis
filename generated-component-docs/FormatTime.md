# FormatTime

# FormatTime

In Jobber a FormatTime is used to ensure that time is displayed in the expected
format. No text styling is applied, this simply formats the text.

## Specify format

You can select a specific format to use in FormatTime.

In the following example, we create a new `Date` object and set its time to
22:35:00 (10:35 PM) using the `setHours` method. The `setHours` method sets the
hours, minutes, seconds, and milliseconds for a specified date.

```tsx
const newDate = new Date();
newDate.setHours(22, 35, 0, 0);

<FormatTime time={newDate} use24HourClock={true} /> // prints 22:35

<FormatTime time={newDate} use24HourClock={false} /> // prints 10:35 PM

```

## Web Component Code

```tsx
FormatTime  Web React import React from "react";

interface FormatTimeProps {
  /**
   * Time (as JS Date) to be displayed.
   *
   * A `string` should be an ISO 8601 format date string.
   */
  readonly time: Date | string;

  /**
   * Optionally specify clock format. If `undefined` system format will be respected.
   */
  readonly use24HourClock?: boolean;
}

export function FormatTime({
  time: inputTime,
  use24HourClock,
}: FormatTimeProps) {
  let dateObject: Date;

  if (inputTime instanceof Date) {
    dateObject = inputTime;
  } else {
    dateObject = new Date(inputTime);
  }

  return <>{dateToLocaleTimeString(dateObject, use24HourClock)}</>;
}

function dateToLocaleTimeString(date: Date, use24HourClock?: boolean) {
  const language = globalThis?.navigator ? navigator.language : "en";

  return date.toLocaleTimeString(language, {
    hourCycle: use24HourClock ? "h23" : "h12",
    minute: "2-digit",
    hour: "numeric",
  });
}

```

## Props

### Web Props

| Prop   | Type    | Required | Default | Description |
| ------ | ------- | -------- | ------- | ----------- | ---------------------------------- |
| `time` | `string | Date`    | ✅      | `_none_`    | Time (as JS Date) to be displayed. |

A `string` should be an ISO 8601 format date string. | | `use24HourClock` |
`boolean` | ❌ | `_none_` | Optionally specify clock format. If `undefined`
system format will be respected. |

## Categories

- Utilities

## Web Test Code

```typescript
FormatTime  Web React Test Testing Jest import React from "react";
import { render } from "@testing-library/react";
import { FormatTime } from "./FormatTime";

describe("FormatTime", () => {
  describe.each(
    Object.entries({
      ISO8601DateString: "2019-03-30T14:30",
      Date: new Date("2019-03-30T14:30"),
    }),
  )("%s", (name, date) => {
    it("renders a FormatTime", () => {
      const { container } = render(<FormatTime time={date} />);
      expect(container.textContent).toBe("2:30 PM");
    });

    it("renders a FormatTime using 24 hour clock", () => {
      const { container } = render(
        <FormatTime time={date} use24HourClock={true} />,
      );
      expect(container.textContent).toBe("14:30");
    });

    it("renders a FormatTime using 12 hour clock", () => {
      const { container } = render(
        <FormatTime time={date} use24HourClock={false} />,
      );
      expect(container.textContent).toBe("2:30 PM");
    });
  });

  it("should render 12:30AM as 00:30 using the 24 hour clock", () => {
    const { container } = render(
      <FormatTime time="2019-03-30T00:30" use24HourClock={true} />,
    );
    expect(container.textContent).toBe("00:30");
  });
});

```

## Component Path

`/components/FormatTime`

---

_Generated on 2025-08-21T17:35:16.361Z_
