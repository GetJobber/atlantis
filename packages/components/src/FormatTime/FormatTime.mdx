---
name: FormatTime
menu: Components
route: /components/time-formatter
showDirectoryLink: true
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { CivilTime } from "@std-proposal/temporal";
import { FormatTime } from ".";

# FormatTime

<ComponentStatus stage="ready" responsive="yes" accessible="yes" />

In Jobber a FormatTime is used to ensure that time is displayed in the expected
format. No text styling is applied, this simply formats the text.

## Simple Example

```ts
import { FormatTime } from "@jobber/components/FormatTime";
```

<Playground>
  <FormatTime time={new CivilTime(2, 35)} />
  -
  <FormatTime time={new CivilTime(18, 35)} />
</Playground>

## Props

<Props of={FormatTime} />

---

## Specify Format

<Playground>
  <FormatTime time={new CivilTime(2, 35)} use24HourClock={true} />
  -
  <FormatTime time={new CivilTime(18, 35)} use24HourClock={true} />
  <br />
  <FormatTime time={new CivilTime(2, 35)} use24HourClock={false} />
  -
  <FormatTime time={new CivilTime(18, 35)} use24HourClock={false} />
</Playground>
