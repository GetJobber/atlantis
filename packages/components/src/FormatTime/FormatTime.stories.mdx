import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
import { CivilTime } from "@std-proposal/temporal";
import { FormatTime } from ".";

<Meta title="Components/Utilities/FormatTime" component={FormatTime} />

# FormatTime

In Jobber a FormatTime is used to ensure that time is displayed in the expected
format. No text styling is applied, this simply formats the text.

```ts
import { FormatTime } from "@jobber/components/FormatTime";
```

<Canvas>
  <Story
    name="FormatTime"
    args={{
      time: new CivilTime(2, 35),
    }}
  >
    {args => (
      <>
        <FormatTime {...args} />
        -
        <FormatTime {...args} time={new CivilTime(18, 35)} />
      </>
    )}
  </Story>
</Canvas>

## Props

<ArgsTable of={FormatTime} story="FormatTime" />

---

## Specify format

<Canvas>
  <FormatTime time={new CivilTime(2, 35)} use24HourClock={true} />
  -
  <FormatTime time={new CivilTime(18, 35)} use24HourClock={true} />
  <br />
  <FormatTime time={new CivilTime(2, 35)} use24HourClock={false} />
  -
  <FormatTime time={new CivilTime(18, 35)} use24HourClock={false} />
</Canvas>
