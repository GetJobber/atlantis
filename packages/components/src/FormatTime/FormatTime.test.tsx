import React from "react";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import { CivilTime } from "@std-proposal/temporal";
import { FormatTime } from "./FormatTime";

afterEach(cleanup);

Object.entries({
  CivilDate: new CivilTime(14, 30),
  ISO8601DateString: "2019-03-30T14:30",
  Date: new Date("2019-03-30T14:30"),
}).forEach(([inputType, value]) => {
  it(`renders a FormatTime from ${inputType}`, () => {
    const tree = renderer.create(<FormatTime time={value} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders a FormatTime from ${inputType} using 24 hour clock`, () => {
    const tree = renderer
      .create(<FormatTime time={value} use24HourClock={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders a FormatTime from ${inputType} using 12 hour clock`, () => {
    const tree = renderer
      .create(<FormatTime time={value} use24HourClock={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
