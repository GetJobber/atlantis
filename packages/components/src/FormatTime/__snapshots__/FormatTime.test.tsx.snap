// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a FormatTime from CivilDate 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from CivilDate using 12 hour clock 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from CivilDate using 24 hour clock 1`] = `"14:30"`;

exports[`renders a FormatTime from Date 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from Date using 12 hour clock 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from Date using 24 hour clock 1`] = `"14:30"`;

exports[`renders a FormatTime from ISO8601DateString 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from ISO8601DateString using 12 hour clock 1`] = `"2:30 PM"`;

exports[`renders a FormatTime from ISO8601DateString using 24 hour clock 1`] = `"14:30"`;
